internal/connector/internal/config/connectors.go

```diff
@@ -51,20 +58,82 @@ func NewConnectorsConfig() *ConnectorsConfig {
 
 func (c *ConnectorsConfig) AddFlags(fs *pflag.FlagSet) {
 	fs.StringArrayVar(&c.ConnectorCatalogDirs, "connector-catalog", c.ConnectorCatalogDirs, "Directory containing connector catalog entries")
+	fs.StringArrayVar(&c.ConnectorMetadataDirs, "connector-metadata", c.ConnectorMetadataDirs, "Directory containing connector metadata configuration files")
 	fs.DurationVar(&c.ConnectorEvalDuration, "connector-eval-duration", c.ConnectorEvalDuration, "Connector eval duration in golang duration format")
 	fs.StringArrayVar(&c.ConnectorEvalOrganizations, "connector-eval-organizations", c.ConnectorEvalOrganizations, "Connector eval organization IDs")
 	fs.BoolVar(&c.ConnectorNamespaceLifecycleAPI, "connector-namespace-lifecycle-api", c.ConnectorNamespaceLifecycleAPI, "Enable APIs to create, update, delete non-eval Namespaces")
 	fs.BoolVar(&c.ConnectorEnableUnassignedConnectors, "connector-enable-unassigned-connectors", c.ConnectorEnableUnassignedConnectors, "Enable support for 'unassigned' state for Connectors")
 }
 
 func (c *ConnectorsConfig) ReadFiles() error {
-	typesLoaded := map[string]string{}
-	var values []ConnectorCatalogEntry
+	// read metadata first to merge with catalog next
+	connectorMetadata, err := c.readConnectorMetadata()
+	if err != nil {
+		return err
+	}
+
+	c.CatalogEntries, err = c.readConnectorCatalog(connectorMetadata)
+	if err != nil {
+		return err
+	}
+
+	remainingIds := len(connectorMetadata)
+	if remainingIds > 0 {
+		ids := make([]string, 0, remainingIds)
+		for id := range connectorMetadata {
+			ids = append(ids, id)
+		}
+		return gherrors.Errorf("Found %d unrecognized connector metadata with ids: %s", remainingIds, ids)
+	}
+
+	glog.Infof("loaded %d connector types", len(c.CatalogEntries))
+
+	return nil
+}
+
+func (c *ConnectorsConfig) readConnectorMetadata() (connectorMetadata map[string]ConnectorMetadata, err error) {
+	connectorMetadata = make(map[string]ConnectorMetadata)
+	for _, dir := range c.ConnectorMetadataDirs {
+		metaDir := shared.BuildFullFilePath(dir)
+		err = files.Walk(metaDir, func(path string, info fs.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			// skip over hidden files and dirs..
+			if info.IsDir() || strings.HasPrefix(path, ".") {
+				return nil
+			}
+
+			glog.Infof("loading connector metadata from file %s", path)
 
+			// Read the file
+			var metas []ConnectorMetadata
+			err = shared.ReadYamlFile(path, &metas)
+			if err != nil {
+				return gherrors.Errorf("error reading connector metadata from %s: %s", path, err)
+			}
+			for _, m := range metas {
+				connectorMetadata[m.ConnectorTypeId] = m
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			return nil, gherrors.Errorf("error listing connector metadata in %s: %s", metaDir, err)
```

Comment:
You should be using `%w` verb instead of `%s` so that the underlaying error is correctly wrapped.